pkg/osquery/tables/xfconf/xfconf.go

```diff
@@ -0,0 +1,164 @@
+			}
+
+			results = append(results, userConfig...)
+		}
+	}
+
+	return results, nil
+}
+
+func (t *XfconfQuerier) getUserConfig(u *user.User, dataQuery string, rowData map[string]string) ([]map[string]string, error) {
+	var results []map[string]string
+
+	// First, get user-specific settings
+	userConfigDir := getUserXfconfDir(u)
+	userConfigRows, err := t.getConfigFromDirectory(userConfigDir, dataQuery, rowData)
+	if err != nil {
+		return nil, fmt.Errorf("error getting config for user %s from directory %s: %w", u.Name, userConfigDir, err)
+	}
+	results = append(results, userConfigRows...)
+
+	// Then, get defaults
+	defaultDirs := getDefaultXfconfDirs()
+	for _, dir := range defaultDirs {
+		defaultConfigRows, err := t.getConfigFromDirectory(dir, dataQuery, rowData)
+		if err != nil {
+			return nil, fmt.Errorf("error getting config for user %s from default directory %s: %w", u.Name, dir, err)
+		}
+		results = append(results, defaultConfigRows...)
+	}
+
+	// Make sure we only include defaults if there isn't already a user-specific setting
+	return deduplicate(results), nil
+}
+
+func (t *XfconfQuerier) getConfigFromDirectory(dir string, dataQuery string, rowData map[string]string) ([]map[string]string, error) {
+	var results []map[string]string
+
+	matches, err := filepath.Glob(filepath.Join(dir, "*.xml"))
+	if err != nil {
+		return nil, fmt.Errorf("could not glob for files in directory %s: %w", dir, err)
+	}
+
+	flattenOpts := []dataflatten.FlattenOpts{
+		dataflatten.WithLogger(t.logger),
+		dataflatten.WithQuery(strings.Split(dataQuery, "/")),
+	}
+
+	for _, match := range matches {
+		flattened, err := parseXfconfXml(match, flattenOpts...)
+		if err != nil {
+			return nil, fmt.Errorf("could not read in xml file %s: %w", match, err)
+		}
+		rowData["channel"] = strings.TrimSuffix(filepath.Base(match), ".xml")
+
+		results = append(results, dataflattentable.ToMap(flattened, dataQuery, rowData)...)
+	}
+
+	return results, nil
+}
+
+// getDefaultXfconfDirs returns the path to xfconf per-channel xml files that contain
+// default settings. It checks for an override, otherwise defaults to /etc/xdg/...
+// See https://docs.xfce.org/xfce/xfce4-session/advanced#files_and_environment_variables.
+func getDefaultXfconfDirs() []string {
+	envDefaultDirsStr := os.Getenv("XDG_CONFIG_DIRS")
+	if envDefaultDirsStr != "" {
+		dirs := strings.Split(envDefaultDirsStr, ":")
+		for i, d := range dirs {
+			dirs[i] = filepath.Join(d, xfconfChannelXmlPath)
+		}
+		return dirs
+	}
+
+	return []string{filepath.Join("/", "etc", "xdg", xfconfChannelXmlPath)}
+}
+
+// getUserXfconfDir returns to xfconf per-channel xml files that contain user-specific
+// settings. It checks for an override via environment variable, otherwise defaults to
+// ~/.config/...
+// See https://docs.xfce.org/xfce/xfce4-session/advanced#files_and_environment_variables.
+func getUserXfconfDir(u *user.User) string {
+	userConfigDir := os.Getenv("XDG_CONFIG_HOME")
```

Comment:
I'm not not totally sure how I feel about the `os.Getenv` here. It's not wrong, but...
1. Do we think this will be set in a way daemons pick up? I suspect not since it feels like this is meant to override `~/.config`, so if it's set, it won't have a user portion.
2. It's basically a constant. (I don't think the ENV can change once the program is launched). So we should be able to read it once early on.

Reply:
1 was kind of what I was worried about. I tested by setting the variable in `/etc/environment` and confirmed that launcher picked it up, but 🤷‍♀️

Reply:
Do you think I should leave this as-is, then, or run `echo $XDG_CONFIG_HOME` as the target user? Or another option?

Reply:
I wouldn't `echo $XDG_CONFIG_HOME` -- adding in an exec just for that feel premature. 

I think we could leave it, or remove it. Your call.

Reply:
I guess leave it, then!